e2e/consul/namespaces_oss.go

```diff
@@ -0,0 +1,386 @@
+// +build !ent
+
+// Nomad OSS ignores Consul Namespace configuration in jobs, these e2e tests
+// verify everything still works and is registered into the "default" namespace,
+// since e2e always uses Consul Enterprise. With Consul OSS, there  are no namespaces.
+// and these tests will not work.
```

Comment:
I love how thorough we're being here by having this set of tests, but given that we run Nomad ENT on nightly I worry these tests aren't going to be run very often and may drift from the Nomad ENT tests. It's a bit more of a lift, but would it possible to parameterize these tests so that we assert the namespaces appropriately depending on Nomad OSS/ENT? That'd save you writing a whole separate set of tests, too.

Reply:
Yeah definitely - I'd like to wait until all four* variants of the tests are written though so it's more clear how to structure things to make them parameterized 

OSS + no ACL (this)
ENT + no ACL
OSS + ACL
ENT + ACL